### PR TITLE
Add documentation for esoteric operators in Data.Bits.Lens and System.FilePath.Lens

### PR DESCRIPTION
**Pull request by Taneb**
Been meaning to do this for the past half a decade, sorry it took me so long! :D

### CODE REVIEW

File: src/System/FilePath/Lens.hs [hunk 1]
```diff
@@ -108,10 +108,18 @@ l <</>= r = l <%= (</> r)
 {-# INLINE (<</>=) #-}
 
 
+-- | Add a path onto the end of the target of a 'Lens' and return the origitnal
```
Comment (glguy) [inline src/System/FilePath/Lens.hs]: original

File: src/System/FilePath/Lens.hs [hunk 2]
```diff
@@ -108,10 +108,18 @@ l <</>= r = l <%= (</> r)
...
+-- value.
+--
+-- When you do not need the result of the operation, ('</>~) is more flexible.
```
Comment (glguy) [inline src/System/FilePath/Lens.hs]: Missing close tick

File: src/System/FilePath/Lens.hs [hunk 3]
```diff
@@ -108,10 +108,18 @@ l <</>= r = l <%= (</> r)
...
 (<<</>~) :: Optical' (->) q ((,)FilePath) s FilePath -> FilePath -> q s (FilePath, s)
 l <<</>~ b = l $ \a -> (a, a </> b)
 {-# INLINE (<<</>~) #-}
 
+-- | Add a path onto the end of a targert of a 'Lens into your monad's state
```
Comment (glguy) [inline src/System/FilePath/Lens.hs]: Missing close tick

Comment (jhenahan) [reply src/System/FilePath/Lens.hs]: Also "target".

File: src/Data/Bits/Lens.hs [hunk 4]
```diff
@@ -159,18 +159,70 @@ l <.&.= b = l <%= (.&. b)
...
+-- bitwise '.&.' with another value, returning the original value (or a
+-- monoidal summary of all the original values).
+--
+-- When you do not need the old value, ('.&.=') is more flexible.
+--
+-- >>> runState (_1 <<.&.= 15) (31,0)
+-- (31,(15,0))
+--
+-- @
+-- ('<<.&.=') :: ('MonadState' s m, 'Bits' a)           => Lens'' s a      -> a -> m a
```
Comment (emilypi) [inline src/Data/Bits/Lens.hs]:
```suggestion
-- ('<<.&.=') :: ('MonadState' s m, 'Bits' a)           => 'Lens'' s a      -> a -> m a
```

File: src/Data/Bits/Lens.hs [hunk 5]
```diff
@@ -159,18 +159,70 @@ l <.&.= b = l <%= (.&. b)
...
+-- ('<<.&.=') :: ('MonadState' s m, 'Bits' a, 'Data.Monoid.Monoid' a) => Traversal'' s a -> a -> m a
```
Comment (emilypi) [inline src/Data/Bits/Lens.hs]:
```suggestion
-- ('<<.&.=') :: ('MonadState' s m, 'Bits' a, 'Data.Monoid.Monoid' a) => 'Traversal'' s a -> a -> m a
```

File: src/Data/Bits/Lens.hs [hunk 6]
```diff
@@ -159,18 +159,70 @@ l <.&.= b = l <%= (.&. b)
...
+-- bitwise '.|.' with another value, returning the original value (or a
+-- monoidal summary of all the original values).
+--
+-- When you do not need the old value, ('.|.=') is more flexible.
+--
+-- >>> runState (_1 <<.|.= 7) (28,0)
+-- (28,(31,0))
+--
+-- @
+-- ('<<.|.=') :: ('MonadState' s m, 'Bits' a)           => Lens'' s a      -> a -> m a
```
Comment (emilypi) [inline src/Data/Bits/Lens.hs]:
```suggestion
-- ('<<.|.=') :: ('MonadState' s m, 'Bits' a)           => 'Lens'' s a      -> a -> m a
```

File: src/Data/Bits/Lens.hs [hunk 7]
```diff
@@ -159,18 +159,70 @@ l <.&.= b = l <%= (.&. b)
...
+-- ('<<.|.=') :: ('MonadState' s m, 'Bits' a, 'Data.Monoid.Monoid' a) => Traversal'' s a -> a -> m a
```
Comment (emilypi) [inline src/Data/Bits/Lens.hs]:
```suggestion
-- ('<<.|.=') :: ('MonadState' s m, 'Bits' a, 'Data.Monoid.Monoid' a) => 'Traversal'' s a -> a -> m a
```

**Comment by ekmett** [comment]
Will fix up comments in repo.

**Pull request merged**